### PR TITLE
Add project search

### LINKED ISSUE #185: Filter projects on dashboard view
## What is this feature about (1 sentence)
There should be a way to search or filter projects on the dashboard.

## Why is it needed? What is the value? For whom do we build it?
The dashboard lists all available projects, which makes it hard to find for a specific project when  `Sorry Cypress` is used to manage multiple projects.

## High-level feature overview
- [x] The dashboard should include a search field.
- [x] When the search field is not empty, the list should only list projects that contain the search in the project id
- [x] When no projects are found, an error message should be displayed

### PR DESCRIPTION
**Pull request by dlavrenuek**
This PR adds a search functionality for projects, some minor restyling (UI love) to improve UX and some code refactoring (mostly newly added components)

### Search functionality implementation
- The dashboard should include a search field.
- When the search field is not empty, the list should only list projects that contain the search in the project id
- When no projects are found, an error message should be displayed

#### Dashboard
dashboard

#### List of projects
list

#### No projects found
not found

### Other changes
- The api GraphQL `Filter` type now also includes a field `like` that is parsed as "string includes" rather than equality
- Added missing dev dependencies to `dashboard/package.json`
- Added [prittier](https://prettier.io/) config for common style rules and the command `yarn format` for automatic code formatting (is not run automatically)
- Added common ui components
- Some code format/naming differs from the current one and is made according to known best practices (I can change it to match the style of other code if needed)
  - New files that `export default` a component are named in uppercase
  - `export default` instead of `export const` for new hooks

### What is not included in this PR
- Refactoring of other parts of the code

Closes #185

### CODE REVIEW

**Comment by jhicken** [comment]
Huzza! This has been on my personal todo list for a while. Thanks for doing this.

**Comment by agoldis** [comment]
@jhicken any chance you can review? I am overwhelmed 😓 Do you have permissions to merge / approve?

File: packages/dashboard/.prettierignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Ignore artifacts:
+/dist
+/node_modules
+/src/generated
```
Comment (agoldis) [inline packages/dashboard/.prettierignore:4]: 👍🏻

**Review by agoldis — APPROVED** [review]

**Comment by agoldis** [comment]
@all-contributors please add @dlavrenuek for code

**Comment by agoldis** [comment]
Thanks a lot @dlavrenuek amazing!

**Comment by jhicken** [comment]
@agoldis Yeah I think I do have merge rights. I have mostly just deferred to you. But yeah absolutely. I can see the mass of stuff you are managing. If another one comes up when I have time. I will make sure to do some reviews and merges.

**Comment by agoldis** [comment]
@jhicken much appreciated!

**Comment by dlavrenuek** [comment]
One thing that I forgot to mention is that the click area of the list item now takes up the whole horizontal space, previously you could only click the text itself

list

**Pull request merged**